packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -604,13 +631,25 @@ export function authorizationServiceBuilder(
         readModelService
       );
 
+      if (
+        delegation &&
+        purpose.consumerId !== organizationId &&
+        !eservice.isClientAccessDelegable
+      ) {
+        throw eserviceNotDelegableForClientAccess(eservice);
+      }
+
+      const agreementConsumerId =
+        delegation && purpose.consumerId !== organizationId
+          ? delegation.delegatorId
+          : organizationId;
       const agreement = await readModelService.getActiveOrSuspendedAgreement(
         eservice.id,
-        organizationId
```

Comment:
I think you can directly use `purpose.consumerId` here without the need to get it from the delegation: the consumerId is always the same with or without delegation